clientgenlib/client.go

```
@@ -0,0 +1,1004 @@
+		pktType = pktAnnounceGrantReq
+		atomic.AddUint64(&cfg.Counters.TotalGenMsgSent, 1)
+		atomic.AddUint64(&cfg.Counters.TotalClientAnnounceReq, 1)
+		cl.CountAnnounceGrantReq++
+		if cl.laststate == curState {
+			atomic.AddUint64(&cfg.Counters.TotalClientAnnounceReqResend, 1)
+		}
+	} else if curState == stateGotGrantAnnounce {
+		cl.stateSem.Release(1)
+		// need to request sync grant
+		if cfg.DebugLogClient || cfg.DebugPrint {
+			log.Infof("GotGrantAnnounce cl %v state %v, reqUnicast MessageSync seq=%d", cl.ClientIP, curState, cl.genSequence)
+		}
+		payload = reqUnicast(ptp.ClockIdentity(cl.index), time.Duration(float64(time.Second)*cfg.DurationSec), ptp.MessageSync)
+		pktType = pktSyncGrantReq
+		atomic.AddUint64(&cfg.Counters.TotalGenMsgSent, 1)
+		atomic.AddUint64(&cfg.Counters.TotalClientSyncReq, 1)
+		cl.CountSyncGrantReq++
+		if cl.laststate == curState {
+			atomic.AddUint64(&cfg.Counters.TotalClientSyncReqResend, 1)
+		}
+	} else if curState == stateGotGrantSync {
+		cl.stateSem.Release(1)
+		// need to request DelayResp grant
+		if cfg.DebugLogClient || cfg.DebugPrint {
+			log.Infof("GotGrantSync cl %v state %v, reqUnicast MessageDelayResp seq=%d", cl.ClientIP, curState, cl.genSequence)
+		}
+		payload = reqUnicast(ptp.ClockIdentity(cl.index), time.Duration(float64(time.Second)*cfg.DurationSec), ptp.MessageDelayResp)
+		pktType = pktDelayRespGrantReq
+		atomic.AddUint64(&cfg.Counters.TotalGenMsgSent, 1)
+		atomic.AddUint64(&cfg.Counters.TotalClientDelayRespReq, 1)
+		cl.CountDelayRespGrantReq++
+		if cl.laststate == curState {
+			atomic.AddUint64(&cfg.Counters.TotalClientDelayRespReqResend, 1)
+		}
+	} else {
+		cl.stateSem.Release(1)
+		cl.CountRetransmitWierdState++
+		log.Infof("Client in wierd state during retransmit %v", curState)
+		return
+	}
+
+	cl.laststate = cl.state
+	payload.SetSequence(cl.genSequence)
+	b, _ := ptp.Bytes(payload)
```

Are you ignoring the error on purpose?

---

Changed it to catch the error.